src/db.rs

```diff
@@ -308,9 +315,15 @@ impl DbInner {
                 SsTableId::Wal(id) => *id,
                 SsTableId::Compacted(_) => return Err(SlateDBError::InvalidDBState),
             };
-            let iter =
-                SstIterator::new_spawn(Rc::clone(&sst), db_inner.table_store.clone(), 1, 256, true)
-                    .await?;
+            let iter = SstIterator::new_spawn(
+                Rc::clone(&sst),
+                db_inner.table_store.clone(),
+                1,
+                256,
+                true,
+                sst.info.row_attributes.clone(),
```

Comment:
Do we need this param in `new_spawn` if we're already passing in the SST? Won't it always have the `sst.info.row_attributes` value?

Reply:
ah nice! I had missed that because of the generics. This will be a massive cleanup 🙏

Reply:
Literally saved 100LOC 🙈